### PR TITLE
My profile DSSMOD-1250

### PR DESCRIPTION
**Pull request by mincho-cb-nytd**

### CODE REVIEW

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js [hunk 1]
```diff
@@ -60,4 +62,128 @@ describe("State Authorized Official Account Request Validations", function () {
...
+        editMyProfile.elements.revisedInformationNumber().should('have.text', '(508) 246-0311')
+        editMyProfile.clickOnSaveEdits();
+        editMyProfile.elements.editsSuccessfulText().should('have.text', 'Edits Successful');
+        editMyProfile.elements.SuccessMsg().contains('You have successfully updated the NYTD account information for');
+        editMyProfile.clickOnContinueBtn();
+        commonPage.verifyUrl('/User/Profile');
+        viewMyProfile.elements.myProfileFirstLastName().should('have.text', "cypress cb")
+        viewMyProfile.elements.phoneNumber().should('have.text', '(508) 246-0311')
+    });
 });
```
Comment (wyntonjones) [inline cypress/e2e/My Profile/CBCentralUserMyProfile.js:189]: After saving i dont see steps that verify the changes made. And then you need to change it back to the original information.

Also a test to make the changes but not save them, there will be a unsaved changes modal.

**Review by wyntonjones — CHANGES_REQUESTED** [review]
I only reviewed the CB central my profile test. Make those corrections and apply them, if necessary to the other specs. Be sure to verify button text in the page object file and if the button brings up a modal, verify the modal text, they can be done in its own test within the spec.

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js [hunk 2]
```diff
@@ -60,4 +62,128 @@ describe("State Authorized Official Account Request Validations", function () {
...
+        editMyProfile.elements.oldInformationNumber().should('have.text', '(508) 246-0311')
+        editMyProfile.elements.revisedInformationText().should('have.text', 'Revised Information');
+        editMyProfile.elements.revisedInformationFName().should('have.text', editMyProfile.firstName)
+        editMyProfile.elements.revisedInformationLName().should('have.text', editMyProfile.lastName)
+        editMyProfile.elements.revisedInformationNumber().should('have.text', editMyProfile.phone)
+        editMyProfile.clickOnSaveEdits();
+        editMyProfile.elements.editsSuccessfulText().should('have.text', 'Edits Successful');
+        editMyProfile.elements.SuccessMsg().contains('You have successfully updated the NYTD account information for');
+        editMyProfile.clickOnContinueBtn();
+        commonPage.verifyUrl('/User/Profile');
```
Comment (wyntonjones) [inline cypress/e2e/My Profile/CBCentralUserMyProfile.js:132]: This should be a new test. "Verify the my profile user information changed" or something like that.

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js [hunk 3]
```diff
@@ -60,4 +62,128 @@ describe("State Authorized Official Account Request Validations", function () {
...
+        editMyProfile.elements.currentUserInfoReceivesEmail().should('have.text', 'Yes');
+        editMyProfile.elements.currentUserInfoPhone().should('have.text', editMyProfile.phone);
+        editMyProfile.elements.currentUserInfoPrimaryRole().should('have.text', 'CB Central Office Staff Member');
+        editMyProfile.elements.currentUserInfoDataExport().should('have.text', 'Yes');
+        editMyProfile.elements.currentUserInfoAccountLocked().should('have.text', 'No');
+        editMyProfile.elements.currentUserInfo().should('have.text', 'This information will not be updated until "Preview Edits" and then "Save Edits" have been clicked');
+        editMyProfile.elements.usernameTextBox().should('be.disabled');
+        editMyProfile.elements.userEmailTextBox().should('be.disabled');
+        editMyProfile.elements.previewEditsButton().should('be.disabled');
+        editMyProfile.enterDefaultUsernamePasswordAndPhone('cypress', 'cb', '5082460311');
```
Comment (wyntonjones) [inline cypress/e2e/My Profile/CBCentralUserMyProfile.js:170]: Make this its own test "Change user information back to original user information"

**Review by wyntonjones — CHANGES_REQUESTED** [review]

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js [hunk 4]
```diff
@@ -60,4 +62,128 @@ describe("State Authorized Official Account Request Validations", function () {
...
     commonPage.navigateBack();
     });
+
+    it("Verify cb central user is able to edit firstname,lastname and phone number for my profile", function () {
+        cy.visit('/User');
+        commonPage.verifyUrl('/User');
+        commonPage.clickOnWelcomeBtn();
+        commonPage.verifyUrl('/User/Profile');
+        commonPage.verifyBreadCrumbs('My Profile');
+        viewMyProfile.elements.myProfileFirstLastName().should('have.text', 'cypress cb')
```
[the remark this answers is not in the document]
Comment (mincho-cb-nytd) [reply cypress/e2e/My Profile/CBCentralUserMyProfile.js:72]: resolved

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js [hunk 5]
```diff
@@ -13,10 +15,10 @@ describe("State Authorized Official Account Request Validations", function () {
     commonPage.clickOnMyProfileSelect();
```
[the remark this answers is not in the document]
Comment (mincho-cb-nytd) [reply cypress/e2e/My Profile/CBCentralUserMyProfile.js:15]: it a a dropdown option thats why is named as select...

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js [hunk 6]
```diff
@@ -60,4 +62,128 @@ describe("State Authorized Official Account Request Validations", function () {
...
+        commonPage.elements.headerH3Text().should('have.text', 'Edit My Profile');
+        editMyProfile.elements.firstnameText().should('have.text', 'First Name *');
+        editMyProfile.elements.firstnameTextBox().should('have.value', 'cypress')
+        editMyProfile.elements.lastnameText().should('have.text', 'Last Name *');
+        editMyProfile.elements.lastnameTextBox().should('have.value', 'cb')
+        editMyProfile.elements.phoneText().should('have.text', 'Phone *');
+        editMyProfile.elements.phoneTextBox().should('have.value', '(508) 246-0311');
+        editMyProfile.elements.userEmailText().should('have.text', 'User Email *');
+        editMyProfile.elements.userEmailTextBox().should('have.value', 'tyler.smith+cypresscb@icf.com');
+        editMyProfile.elements.receivesEmailNotificationsText().should('have.text', 'Receive Email Notifications? *');
```
[the remark this answers is not in the document]
Comment (mincho-cb-nytd) [reply cypress/e2e/My Profile/CBCentralUserMyProfile.js:92]: resolved

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js [hunk 7]
```diff
@@ -60,4 +62,128 @@ describe("State Authorized Official Account Request Validations", function () {
...
+        viewMyProfile.elements.deleteMyAccountBtn().should('have.text', 'Delete My Account')
+        viewMyProfile.clickOnEditMyProfileButton();
+        commonPage.verifyUrl('/User/Profile/Edit');
+        commonPage.verifyBreadCrumbs('My Profile', 'Edit My Profile');
+        commonPage.elements.headerH3Text().should('have.text', 'Edit My Profile');
+        editMyProfile.elements.firstnameText().should('have.text', 'First Name *');
+        editMyProfile.elements.firstnameTextBox().should('have.value', 'cypress')
+        editMyProfile.elements.lastnameText().should('have.text', 'Last Name *');
+        editMyProfile.elements.lastnameTextBox().should('have.value', 'cb')
+        editMyProfile.elements.phoneText().should('have.text', 'Phone *');
```
[the remark this answers is not in the document]
Comment (mincho-cb-nytd) [reply cypress/e2e/My Profile/CBCentralUserMyProfile.js:88]: Added step to verify username text box, Unable to too add verification for username text due to button is disabled

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js [hunk 8]
```diff
@@ -60,4 +62,128 @@ describe("State Authorized Official Account Request Validations", function () {
...
+        editMyProfile.elements.phoneTextBox().should('have.value', '(508) 246-0311');
+        editMyProfile.elements.userEmailText().should('have.text', 'User Email *');
+        editMyProfile.elements.userEmailTextBox().should('have.value', 'tyler.smith+cypresscb@icf.com');
+        editMyProfile.elements.receivesEmailNotificationsText().should('have.text', 'Receive Email Notifications? *');
+        editMyProfile.elements.primaryRoleText().should('have.text', 'Primary Role');
```
[the remark this answers is not in the document]
Comment (mincho-cb-nytd) [reply cypress/e2e/My Profile/CBCentralUserMyProfile.js:93]: done

File: cypress/e2e/pages/EditMyProfilePageObjects.js [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    enterFirstName(){
+        this.elements.firstnameTextBox().clear().type(this.firstName);
+    }
+    enterLastName() {
+        this.elements.lastnameTextBox().clear().type(this.lastName);
+    }
+    enterPhoneNumber() {
+        this.elements.phoneTextBox().clear().type(this.phone);
+    }
+    clickOnPreviewEdits() {
```
[the remark this answers is not in the document]
Comment (mincho-cb-nytd) [reply cypress/e2e/pages/EditMyProfilePageObjects.js:94]: this is already verify on line number 20

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js [hunk 10]
```diff
@@ -60,4 +62,128 @@ describe("State Authorized Official Account Request Validations", function () {
...
+        editMyProfile.elements.systemAdministratorRadioButton().should('have.text', 'System Administrator');
+        editMyProfile.elements.cbCentralOfficeStaffMemberRadioButton().should('have.text', 'CB Central Office Staff Member');
+        editMyProfile.elements.regionalOfficeUserRadioButton().should('have.text', 'Regional Office User');
+        editMyProfile.elements.additionalPermissionsText().should('have.text', 'Additional Permissions');
+        editMyProfile.elements.dataExportCheckBox().should('have.text', 'Data Export');
+        editMyProfile.elements.previewEditsButton().should('have.text', 'Preview Edits');
```
[the remark this answers is not in the document]
Comment (mincho-cb-nytd) [reply cypress/e2e/My Profile/CBCentralUserMyProfile.js:99]: done

File: cypress/e2e/pages/EditMyProfilePageObjects.js [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+        this.elements.previewEditsButton().click();
+    }
+
+    clickOnSaveEdits() {
```
[the remark this answers is not in the document]
Comment (mincho-cb-nytd) [reply cypress/e2e/pages/EditMyProfilePageObjects.js:98]: line 133 verifying the text

File: cypress/e2e/My Profile/CBCentralUserMyProfile.js (hunk 3 above)
Comment (mincho-cb-nytd) [reply cypress/e2e/My Profile/CBCentralUserMyProfile.js:170]: resolved

**Review by TylerFSmith2 — COMMENTED** [review]
I did not mark every instance of a mng locator, please be sure to replace all of them.

**Pull request merged**